index.js

```diff
@@ -141,6 +141,14 @@ var compile = function(schema, cache, root, reporter, opts) {
       }
     }
 
+    if (node.default !== undefined && opts && opts.setDefault === true) {
+      validate
+        ('if (%s === undefined) {', name)
+          ('console.log(%s)', JSON.stringify(name))
```

Comment:
Sorry about that `console.log`... Pretty sad that I managed to somehow slip that into a 6 line patch :)
After all everything else is just tests and docs.